jenkins/python-setup.sh

```diff
@@ -17,10 +17,22 @@
 # chosen over any installed ones, and it adds `/usr/sbin` to the end of the
 # path for the `ip` command, used by pbench-register-tool.
 
-PATH=$(python3 -m site --user-base)/bin:${PATH}:/usr/sbin
+function _pbench_pathins() {
+    if [[ ":${PATH:-}:" != *":${1}:"* ]]; then
```

Comment:
Does the `:-` add anything here?  (If `PATH` is empty or undefined, then `${PATH}` will expand vacuously which I think is the same result as what you have there.)

Reply:
Yes, this is tried and true code from `agent/profile`.

Reply:
What does it add (other than complication and confusion for the reader)?

Reply:
Well, then, I propose that we omit it and go with
```{
    if [[ ":${PATH}:" != *":${1}:"* ]]; then
```

Reply:
Have you tried that?  Could you propose that change for `agent/profile` and ensure it works before we try another experiment here?  It would really help the whole team to move on this work.

The reason for using the existing `agent/profile` method is to make sure we are properly adding the paths without losing the existing `PATH` found in the container and without adding an unintentionally occluded path entry.

Reply:
> Have you tried that?

Interactively, sure...and it works as documented and expected.

> The reason for using the existing `agent/profile` method is [...]

I understand what the code is endeavoring to do; I'm objecting to a detail of how it is doing it.  Moreover, if we're so intent on doing the same thing here that we do in `agent/profile`, then we should be sharing that code, not copying it...then we can fix it in one place....

Reply:
If you can propose a way to share the code, perhaps a PR would be helpful?

I have noodled about this, and I don't know how to share the `agent/profile` for those functions in the this file without going through contortions that would affect the `agent` code.  So this seemed like a good case to violate the `DRY` principle.

As for dropping the `:-`, if that will allow this PR to get merged, I'll make the change.

Reply:
I believe in Bash the `source` command can function like `import` does in Python, allowing you to pull in function definitions from external files...so we could build a "library" of useful Bash functions and pull it in to our favorite scripts (sort of like how we execute `agent/base`, now).

Alternatively, we could contemplate executing `agent/profile` (perhaps with some parameterization) from this script, since, in both cases, we seem to be trying to accomplish the same things.

But, as you say, that's for another PR.